framework/src/application/node/processor/processor.js

```diff
@@ -268,121 +265,88 @@ class Processor {
 	async applyGenesisBlock(block) {
 		this.logger.info({ id: block.id }, 'Applying genesis block');
 		const blockProcessor = this._getBlockProcessor(block);
-		return this._processGenesis(block, blockProcessor, { skipSave: true });
+		return this._processGenesis(block, blockProcessor, { saveOnlyState: true });
 	}
 
 	async _processValidated(
 		block,
 		lastBlock,
 		processor,
-		{ skipSave, skipBroadcast, removeFromTempTable = false } = {},
+		{ saveOnlyState, skipBroadcast, removeFromTempTable = false } = {},
 	) {
-		await this.storage.entities.Block.begin(
-			'App:Node:processBlock',
-			async tx => {
-				const stateStore = new StateStore(this.storage, { tx });
-				// initialize chain state
-				await stateStore.chainState.cache();
-
-				await processor.verify.run({
-					block,
-					lastBlock,
-					skipExistingCheck: skipSave,
-					stateStore,
-					tx,
-				});
+		const stateStore = this.blocksModule.newStateStore();
 
-				const blockJSON = await this.serialize(block);
-				if (!skipBroadcast) {
-					this.channel.publish('app:processor:broadcast', {
-						block: blockJSON,
-					});
-				}
+		await processor.verify.run({
+			block,
+			lastBlock,
+			skipExistingCheck: saveOnlyState,
+			stateStore,
+		});
 
-				if (!skipSave) {
-					// TODO: After moving everything to state store, save should get the state store and finalize the state store
-					await this.blocksModule.save(blockJSON, tx);
-				}
+		const blockJSON = await this.serialize(block);
+		if (!skipBroadcast) {
+			this.channel.publish('app:processor:broadcast', {
+				block: blockJSON,
+			});
+		}
 
-				// Apply should always be executed after save as it performs database calculations
-				// i.e. Dpos.apply expects to have this processing block in the database
-				await processor.apply.run({
-					block,
-					lastBlock,
-					skipExistingCheck: skipSave,
-					stateStore,
-					tx,
-				});
+		// Apply should always be executed after save as it performs database calculations
+		// i.e. Dpos.apply expects to have this processing block in the database
+		await processor.apply.run({
+			block,
+			lastBlock,
+			skipExistingCheck: saveOnlyState,
+			stateStore,
+		});
 
-				if (removeFromTempTable) {
-					await this.blocksModule.removeBlockFromTempTable(block.id, tx);
-					this.logger.debug(
-						{ id: block.id, height: block.height },
-						'Removed block from temp_blocks table',
-					);
-				}
+		await this.blocksModule.save(block, stateStore, {
+			saveOnlyState,
+			removeFromTempTable,
+		});
 
-				// Should only publish 'app:processor:newBlock' if saved AND applied successfully
-				if (!skipSave) {
-					this.channel.publish('app:processor:newBlock', {
-						block: blockJSON,
-					});
-				}
+		// Should only publish 'app:processor:newBlock' if saved AND applied successfully
+		if (!saveOnlyState) {
+			this.channel.publish('app:processor:newBlock', {
+				block: blockJSON,
+			});
+		}
 
-				return block;
-			},
-		);
+		return block;
```

Comment:
do we use the return block? or should this function return void?

Reply:
In some case, it might need processed block